### PR TITLE
Refactoring/#203 clean package names and directory structure

### LINKED ISSUE #203: Clean package names and directory structure
[Comment from @Nicoretti](https://github.com/exasol/advanced-analytics-framework/pull/202/files#r1810784044) on PR #202 

```python 
from exasol_advanced_analytics_framework.udf_framework.query_handler_runner_udf \
    import QueryHandlerRunnerUDF
```

Naming is slightly out of control, the full name contains:
* 3x  udf
* 2x query
* 2x framework
* 2x handler
* 2x runner

which results in the need to break the import line ...

### PR DESCRIPTION
**Pull request by ckunki**
Closes #203

### CODE REVIEW

File: tests/unit_tests/query_handler_runner/test_python_query_handler_runner.py
Comment (ckunki) [inline tests/unit_tests/query_handler_runner/test_python_query_handler_runner.py:1]: Please take a closer look at this file.
All other files are only changed wrt. to renamed packages and directory structure. 

This file needed additional changes as file mock_sql-executor was replaced which required to change the implementation of most of the test cases in this file.

File: advanced-analytics-framework-0.1.0-1.rockspec [hunk 1]
```diff
@@ -23,10 +23,11 @@ dependencies = {
...
    modules = {
-      ["exasol.advanced_analytics_framework.exasol_script_tools"] = "exasol_advanced_analytics_framework/lua/src/exasol_script_tools.lua",
-      ["exasol.advanced_analytics_framework.query_loop"] = "exasol_advanced_analytics_framework/lua/src/query_loop.lua",
-      ["exasol.advanced_analytics_framework.query_handler_runner_main"] = "exasol_advanced_analytics_framework/lua/src/query_handler_runner_main.lua",
-      ["exasol.advanced_analytics_framework.query_handler_runner"] = "exasol_advanced_analytics_framework/lua/src/query_handler_runner.lua"
+      -- should the lua rock modules be renamed, too?
+      ["exasol.advanced_analytics_framework.exasol_script_tools"] = "exasol/analytics/lua/src/exasol_script_tools.lua",
+      ["exasol.advanced_analytics_framework.query_loop"] = "exasol/analytics/lua/src/query_loop.lua",
+      ["exasol.advanced_analytics_framework.query_handler_runner_main"] = "exasol/analytics/lua/src/query_handler_runner_main.lua",
+      ["exasol.advanced_analytics_framework.query_handler_runner"] = "exasol/analytics/lua/src/query_handler_runner.lua"
```
Comment (ckunki) [inline advanced-analytics-framework-0.1.0-1.rockspec:30]: Should the lua rock modules be renamed, too?

File: exasol/analytics/query/handler/query/drop/interface.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from exasol.analytics.query.handler.query.interface import Query
+
+
+class DropyQuery(Query):
+    pass
+
+
+class DropQuery(Query):
+    pass
```
Comment (Nicoretti) [inline exasol/analytics/query/handler/query/drop/interface.py:9]: Duplicated?

File: exasol/analytics/query/handler/udf/interface.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+from abc import ABC, abstractmethod
+
+from exasol.analytics.query.handler.context.scope import     ScopeQueryHandlerContext
+
+
+class UDFQueryHandler(QueryHandler[str, str]):
+    """Abstract class for QueryHandlers used in QueryHandlerRunnerUDF"""
+
+
+class UDFQueryHandlerFactory(ABC):
```
Comment (Nicoretti) [inline exasol/analytics/query/handler/udf/interface.py:13]:
```suggestion
class Factory(ABC):
```

File: exasol/analytics/query/handler/udf/interface.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+from exasol.analytics.query.handler.query_handler import QueryHandler
+
+
+from abc import ABC, abstractmethod
+
+from exasol.analytics.query.handler.context.scope import     ScopeQueryHandlerContext
+
+
+class UDFQueryHandler(QueryHandler[str, str]):
```
Comment (Nicoretti) [inline exasol/analytics/query/handler/udf/interface.py:9]:
```suggestion
class Handler(QueryHandler[str, str]):
```

File: exasol/analytics/query/handler/udf/runner/state.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    Column
+
+from exasol.analytics.query.handler.context.scope import     ScopeQueryHandlerContext
+from exasol.analytics.query.handler.context.top_level_query_handler_context import     TopLevelQueryHandlerContext
+from exasol.analytics.query.handler.query_handler     import QueryHandler
+from exasol.analytics.query.handler.udf.connection_lookup import UDFConnectionLookup
+
+
+@dataclass()
+class QueryHandlerRunnerState:
```
Comment (Nicoretti) [inline exasol/analytics/query/handler/udf/runner/state.py:14]:
```suggestion
class State:
```

File: exasol/analytics/schema/__init__.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+from .view_name_impl import ViewNameImpl
+from .view import View
+
+from .view_name_builder import ViewNameBuilder
+from .udf_name_builder import UDFNameBuilder
+from .table_name_builder import TableNameBuilder
+from .column_builder import ColumnBuilder
+from .column_name_builder import ColumnNameBuilder
+from .table_builder import TableBuilder
+from .connection_object_name_builder import ConnectionObjectNameBuilder
```
Comment (Nicoretti) [inline exasol/analytics/schema/__init__.py:37]: why relative instead of fully qualified imports?

**Review by Nicoretti — COMMENTED** [review]
I am a huge fan of your refactoring :+1: , in the long run you may consider cleaning it up further e.g. by:
Changing:
```python
from exasol.analytics.deployment.lua_script_bundle import LuaScriptBundle
```
To:
```python
from exasol.analytics.deployment.lua import ScriptBundle
```
etc.

File: exasol/analytics/schema/__init__.py (hunk 6 above)
Comment (ckunki) [reply exasol/analytics/schema/__init__.py]: Thanks. See next push using fully qualified imports.

File: exasol/analytics/query/handler/context/query_handler_context.py
[the remark this answers is not in the document]
Comment (ckunki) [reply exasol/analytics/query/handler/context/query_handler_context.py:1]: There is a query_handler_context.py and a top_level_query_handler_context.py.

File: advanced-analytics-framework-0.1.0-1.rockspec (hunk 1 above)
Comment (tkilias) [reply advanced-analytics-framework-0.1.0-1.rockspec:30]: If, then in another PR. And, probably later. I don't want to waste time with that at the moment

File: exasol/analytics/query/handler/udf/runner/state.py (hunk 5 above)
Comment (ckunki) [reply exasol/analytics/query/handler/udf/runner/state.py:14]: The current PR only renames files and directories. 
I propose to rename classes in the scope of the separate issue #211.

File: exasol/analytics/query/handler/udf/interface.py (hunk 4 above)
Comment (ckunki) [reply exasol/analytics/query/handler/udf/interface.py:9]: The current PR only renames files and directories. 
I propose to rename classes in the scope of the separate issue #211.

File: exasol/analytics/query/handler/udf/interface.py (hunk 3 above)
Comment (ckunki) [reply exasol/analytics/query/handler/udf/interface.py:13]: The current PR only renames files and directories. 
I propose to rename classes in the scope of the separate issue #211.

File: exasol/analytics/query/handler/query/drop/interface.py (hunk 2 above)
Comment (ckunki) [reply exasol/analytics/query/handler/query/drop/interface.py]: Haha -- the first class contains an extra `y`: Dop-y-Query :rofl:

File: exasol/analytics/query/handler/context/proxy/udf_name_proxy.py
[the remark this answers is not in the document]
Comment (ckunki) [reply exasol/analytics/query/handler/context/proxy/udf_name_proxy.py:1]: Good proposal!
Please see next push.

File: exasol/analytics/query/handler/context/proxy/table_name_proxy.py
[the remark this answers is not in the document]
Comment (ckunki) [reply exasol/analytics/query/handler/context/proxy/table_name_proxy.py:1]: Good proposal!
Please see next push.

**Pull request merged**